tap_square/client.py

```diff
@@ -86,8 +86,17 @@ def _retryable_v2_method(request_method, body, **kwargs):
         result = request_method(body, **kwargs)
 
         if result.is_error():
+            LOGGER.info("HTTP status code when it errors out: %s", result.status_code)
             error_message = result.errors if result.errors else result.body
-            if 'Service Unavailable' in error_message or 'upstream connect error or disconnect/reset before headers' in error_message or result.status_code == 429:
+
+            # Refactor the conditions into separate variables for readability
+            is_service_unavailable = 'Service Unavailable' in error_message
+            is_upstream_error = 'upstream connect error or disconnect/reset before headers' in error_message
+            is_cf_error_1101 = '<span class="cf-error-code">1101</span>' in error_message
+            is_html_error = error_message.startswith('<!DOCTYPE html>')
+            is_status_429_or_500 = result.status_code == 429 or result.status_code >= 500
+
+            if is_service_unavailable or is_upstream_error or is_cf_error_1101 or is_html_error or is_status_429_or_500:
                 raise RetryableError(error_message)
             else:
                 raise RuntimeError(error_message)
```

Comment:
```suggestion
            if any(is_service_unavailable, is_upstream_error, is_cf_error_1101, is_html_error, is_status_429_or_500):
                raise RetryableError(error_message)
            else:
                raise RuntimeError(error_message)
```

Reply:
The any() function expects an iterable (like a list, set, or tuple). Hence, made the appropriate change in the code.

Reply:
You are right, that was a typo. Corrected!